src/main/java/org/broadinstitute/hellbender/engine/spark/datasources/ReadsSparkSource.java

```diff
@@ -107,10 +115,23 @@ public ReadsSparkSource(JavaSparkContext ctx) {
     /**
      * Loads the header using Hadoop-BAM.
      * @param filePath path to the bam.
+     * @param auth authentication information if using GCS.
      * @return the header for the bam.
      */
-    public static SAMFileHeader getHeader(final JavaSparkContext ctx, final String filePath) {
+    public static SAMFileHeader getHeader(final JavaSparkContext ctx, final String filePath, final AuthHolder auth) {
         final SAMFileHeader samFileHeader;
+
+        // GCS case
+        if (BucketUtils.isCloudStorageUrl(filePath)) {
```

Comment:
This shouldn't be necessary if using the hdfs -> GCS adapter? Can you explain why you added this, given that you're not using this class at all?